### DISCUSSION TITLE
Content Too Large for FormData

### DISCUSSION THREAD

**Discussion by zAlweNy26** [johannschopplich/nuxt-api-party#63] [Q&A] [answered]
Hi,

first of all, thanks for the amazing module, it works like a charm and it sped up my DX a lot.
Just one thing I don't understand (if I'm doing wrong or it's a bug), and that thing is `FormData`.
When doing a request with the `$myApi` like this:
```ts
const formData = new FormData()
formData.append('file', file, file.name)
$myApi('my_endpoint/path', {
	method: 'POST',
	body: formData
})
```
It works for small files (like smaller than 1 MB, but I don't know precisely). But If I want to send a big file (for example, 10 MB), I get the error code `413 Content Too Large` and an internal error in the server console:
```
 ERROR  [unhandledRejection] write EPIPE  

  at afterWriteDispatched (node:internal/stream_base_commons:160:15)
  at writeGeneric (node:internal/stream_base_commons:151:3)
  at Socket._writeGeneric (node:net:952:11)
  at Socket._write (node:net:964:8)
  at doWrite (node:internal/streams/writable:582:12)
  at clearBuffer (node:internal/streams/writable:765:7)
  at Writable.uncork (node:internal/streams/writable:515:7)
  at connectionCorkNT (node:_http_outgoing:955:8)
  at process.processTicksAndRejections (node:internal/process/task_queues:81:21)

```

And I don't know how to fix this. Do you have any suggestions?

**Comment by johannschopplich** [comment]
What deployment provider do you use? To get around your provider's restrictions, you can enable [client-side API requests](https://nuxt-api-party.byjohann.dev/api/my-api.html#client-requests). Note that this will expose your API credentials to the frontend.

As the form data is serialised for the internal API proxy, its size increases by 33%. Perhaps your provider has a limit on how large the content can be?

**Comment by zAlweNy26** [comment]
I have a backend in PHP made with Symfony. I already set the limit to 20 MB and it works fine from the Swagger docs, but not from my frontend. I can't do a client-side request like you said, because I have some credentials to hide. Do you know what could it be?

**Reply by johannschopplich** [reply]
I didn't mean your PHP backend, but your Nuxt deployment provider. As stated in the [how it works guide](https://nuxt-api-party.byjohann.dev/guide/how-it-works.html), this module will provide an internal Nuxt API route to proxy your request. The error originates in the Node part of your application (frontend), not backend.

**Reply by zAlweNy26** [reply]
Sorry I'm probably too stupid for this, but what do you mean with `Nuxt deployment provider`?
I'm building with SSR and run the dev server in localhost with `nuxi dev`. I understood how this module works, I was using a custom proxy before discovering this package. Where can I set the limit in the node part of my application?

**Reply by johannschopplich** [reply]
Sorry, so if you're developing locally, then your error 413 Content Too Large is probably thrown by your backend. I thought you had deployed your application. Forget I mentioned it. 😄 

If you provide a reproducible example, I can look into it. Otherwise, I have too little information about your setup, where the error originates or how large the content actually is. You can try to increase your file limit. Maybe that will solve your problem.

**Reply by zAlweNy26** [reply, marked as answer]
Yeah at the moment I'm obviously testing it locally, thanks for the help, I forgot that the `nuxt-security` module has a request size limiter...

**Reply by johannschopplich** [reply]
I see!
